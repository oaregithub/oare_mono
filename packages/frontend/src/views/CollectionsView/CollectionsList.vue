<template>
  <div>
    <div class="fixed">
      <v-btn
        v-for="(lett, lettGroup) in letterGroups"
        class="mr-2 mb-2"
        :key="lettGroup"
        fab
        small
        color="primary"
        :to="`/collections/${encodedLetter(lettGroup)}`"
        >{{ lettGroup }}</v-btn
      >
    </div>
    <div v-for="collection in collections" :key="collection.uuid">
      <router-link :to="`/collections/name/${collection.uuid}`"
        >{{ collection.name }}
      </router-link>
    </div>
  </div>
</template>

<script lang="ts">
import {
  defineComponent,
  ref,
  Ref,
  onMounted,
  computed,
  PropType,
} from '@vue/composition-api';
import { CollectionListItem } from '@oare/types';
import { letterGroups } from './utils';

export default defineComponent({
  name: 'CollectionsList',
  props: {
    collections: {
      type: Array as PropType<CollectionListItem[]>,
      required: true,
    },
  },
  setup() {
    return {
      letterGroups,
      encodedLetter: (letter: string) => encodeURIComponent(letter),
    };
  },
});
</script>

<style></style>
