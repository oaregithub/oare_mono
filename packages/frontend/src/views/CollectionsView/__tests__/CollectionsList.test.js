import Vuetify from 'vuetify';
import VueCompositionApi from '@vue/composition-api';
import { createLocalVue } from '@vue/test-utils';
import { render, fireEvent } from '@testing-library/vue';
import CollectionsList from '../CollectionsList.vue';

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(VueCompositionApi);

describe('CollectionsList test', () => {
  const createWrapper = () =>
    render(CollectionsList, {
      localVue,
      vuetify,
      stubs: ['router-link'],
      propsData: {
        collections: [
          {
            uuid: '1',
            name: 'Adana Museum',
          },
          {
            uuid: '2',
            name: 'Istanbul Archaeological Museum',
          },
          {
            uuid: '3',
            name: 'Kayseri Museum',
          },
          {
            uuid: '4',
            name: 'Kt 08/k',
          },
          {
            uuid: '5',
            name: 'Metropolitan Museum of Art',
          },
          {
            uuid: '6',
            name: 'Schoyen Collection',
          },
        ],
      },
    });

  it('matches snapshot', () => {
    expect(createWrapper()).toMatchSnapshot();
  });

  it('displays A-J texts', () => {
    const { getByText } = createWrapper();
    ['Adana Museum', 'Istanbul Archaeological Museum'].forEach((name) => {
      expect(getByText(name));
    });
  });

  it('displays K texts', async () => {
    const { getByText } = createWrapper();
    const kButton = getByText('K');

    await fireEvent.click(kButton);
    [('Kayseri Museum', 'Kt 08/k')].forEach((name) => {
      expect(getByText(name));
    });
  });

  it('displays L-Z texts', async () => {
    const { getByText } = createWrapper();
    const lzButton = getByText('L-Z');

    await fireEvent.click(lzButton);
    [('Metropolitan Museum of Art', 'Schoyen Collection')].forEach((name) => {
      expect(getByText(name));
    });
  });
});
