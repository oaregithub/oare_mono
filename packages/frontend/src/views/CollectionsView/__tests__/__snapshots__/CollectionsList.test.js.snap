// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsList test matches snapshot 1`] = `
Object {
  "baseElement": <body>
    <div
      data-app="true"
    />
    <div>
      <div>
        <div
          class="fixed"
        >
          <router-link-stub
            class="mr-2 mb-2 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round v-btn--router theme--light v-size--small primary"
          >
            <span
              class="v-btn__content"
            >
              A-J
            </span>
          </router-link-stub>
          <router-link-stub
            class="mr-2 mb-2 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round v-btn--router theme--light v-size--small primary"
          >
            <span
              class="v-btn__content"
            >
              K
            </span>
          </router-link-stub>
          <router-link-stub
            class="mr-2 mb-2 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round v-btn--router theme--light v-size--small primary"
          >
            <span
              class="v-btn__content"
            >
              L-Z
            </span>
          </router-link-stub>
        </div>
         
        <div>
          <router-link-stub
            to="/collections/name/1"
          >
            Adana Museum
    
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/collections/name/2"
          >
            Istanbul Archaeological Museum
    
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/collections/name/3"
          >
            Kayseri Museum
    
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/collections/name/4"
          >
            Kt 08/k
    
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/collections/name/5"
          >
            Metropolitan Museum of Art
    
          </router-link-stub>
        </div>
        <div>
          <router-link-stub
            to="/collections/name/6"
          >
            Schoyen Collection
    
          </router-link-stub>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="fixed"
      >
        <router-link-stub
          class="mr-2 mb-2 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round v-btn--router theme--light v-size--small primary"
        >
          <span
            class="v-btn__content"
          >
            A-J
          </span>
        </router-link-stub>
        <router-link-stub
          class="mr-2 mb-2 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round v-btn--router theme--light v-size--small primary"
        >
          <span
            class="v-btn__content"
          >
            K
          </span>
        </router-link-stub>
        <router-link-stub
          class="mr-2 mb-2 v-btn v-btn--is-elevated v-btn--fab v-btn--has-bg v-btn--round v-btn--router theme--light v-size--small primary"
        >
          <span
            class="v-btn__content"
          >
            L-Z
          </span>
        </router-link-stub>
      </div>
       
      <div>
        <router-link-stub
          to="/collections/name/1"
        >
          Adana Museum
    
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/collections/name/2"
        >
          Istanbul Archaeological Museum
    
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/collections/name/3"
        >
          Kayseri Museum
    
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/collections/name/4"
        >
          Kt 08/k
    
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/collections/name/5"
        >
          Metropolitan Museum of Art
    
        </router-link-stub>
      </div>
      <div>
        <router-link-stub
          to="/collections/name/6"
        >
          Schoyen Collection
    
        </router-link-stub>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "isUnmounted": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "unmount": [Function],
  "updateProps": [Function],
}
`;
